src/script/mls/MLSConversations.test.ts

```diff
@@ -0,0 +1,118 @@
+/*
+ * Wire
+ * Copyright (C) 2018 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+import {ConversationProtocol, CONVERSATION_TYPE} from '@wireapp/api-client/lib/conversation';
+
+import {randomUUID} from 'crypto';
+
+import {Account} from '@wireapp/core';
+
+import {initMLSConversations, registerUninitializedConversations} from './MLSConversations';
+import {mlsConversationState} from './mlsConversationState';
+
+import {Conversation} from '../entity/Conversation';
+import {User} from '../entity/User';
+
+function createConversation(protocol: ConversationProtocol) {
+  const conversation = new Conversation(randomUUID(), '', protocol);
+  if (protocol === ConversationProtocol.MLS) {
+    conversation.groupId = `groupid-${randomUUID()}`;
+  }
+  return conversation;
+}
+
+function createConversations(nbConversations: number, protocol: ConversationProtocol = ConversationProtocol.MLS) {
+  return Array.from(new Array(nbConversations)).map(() => createConversation(protocol));
+}
+
+function mockCore() {
+  return {
+    configureMLSCallbacks: jest.fn(),
+    service: {mls: {registerConversation: jest.fn()}},
+  } as unknown as Account;
+}
+
+describe('MLSConversations', () => {
+  beforeEach(() => {
+    jest.spyOn(mlsConversationState.getState(), 'sendExternalToPendingJoin').mockReturnValue(undefined);
+  });
+
+  describe('initMLSConversations', () => {
+    it('joins all the MLS conversations', async () => {
+      const nbProteusConversations = 5 + Math.ceil(Math.random() * 10);
```

Comment:
Since the variable says `nbProteusConversations` I believe it's explicit enough that it's the number of conversations that I want to create. Don't you think?